### ISSUE TITLE
Rollup canonicalization handling

### ISSUE THREAD

**Issue by edoardocavazza** [systemjs/builder#476]
Hi,

I have troubles trying to import a third party library in a SFX bundle. The project structure is:

src/myClass.js
```js
import Lib from 'external-lib'
```

node_modules/external-lib/external-lib.js
```js
module.exports = function() { ... }
```
system.config.js
```js
(function () {
    var isNode = typeof window === 'undefined';

    System.config({
        baseURL: '.',
        transpiler: 'plugin-babel',
        defaultJSExtensions: true,
        format: 'esm',
        packages: {
            './node_modules/external-lib': {
                main: 'external-library.js',
                format: 'cjs'
            },
        },
        paths: {
            'plugin-babel': './node_modules/systemjs-plugin-babel/plugin-babel.js',
            'systemjs-babel-build': './node_modules/systemjs-plugin-babel/systemjs-babel-' + (isNode ? 'node' : 'browser') + '.js',
            'external-lib': './node_modules/external-lib',
        }
    });
})();

```
---

The output is:
```js
$__System.registerDynamic("2", [], true, function($__require, exports, module) {
    // Library body
    return module.exports;
});

$__System.register("1", ["external-lib"], function($__export) {
    // myClass body
});
```

It fails because the external library has been registered as module `"2"`, while the dependency of `"1"` is still `"external-lib"`.

Am I missing something?

Really thanks for your job!

**Comment by guybedford** [comment]
Which version of SystemJS builder are you using here? Is there anyway you can share an isolated replication of the build you ran to get this output?

**Comment by edoardocavazza** [comment]
SystemJS Builder version: 0.15.3

Here the sample: https://plnkr.co/edit/lLROSd38oMzlTwfBpcRa

Thanks @guybedford

**Comment by guybedford** [comment]
@edoardocavazza do you have a test case of how you created this build file?

**Comment by edoardocavazza** [comment]
@guybedford I have created a repo from the plunker above: https://github.com/edoardocavazza/systemjs-test-case

You can run `npm install && node build.js` to bundle the source.

**Comment by guybedford** [comment]
Thanks so much, that helped to track this down.

It turns out this is something a little more general which I will track here, which is that the rollup process needs to recanonicalize the dependencies in https://github.com/systemjs/builder/blob/master/lib/rollup.js#L274, and include the normalized depMap with externals.

To work around the issue set `rollup: false` in the build options.

**Comment by guybedford** [comment]
Released in 0.15.8.
